### PR TITLE
Uses MappedByteBuffer for io, trie instead of map and parallelStream

### PR DESCRIPTION
**Pull request by hallvard**
#### Check List:
- [x ] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [ x] Output matches that of `calculate_average_baseline.sh`
* Execution time: 21 sec
* Execution time of reference implementation: 3 min, 8 sec

### CODE REVIEW

File: src/main/java/dev/morling/onebrc/CalculateAverage_hallvard.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
+package dev.morling.onebrc;
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_hallvard.java:1]: Can you please run `mvn license:format` to add the missing header? Thx!

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
See above.

**Comment by hallvard** [comment]
I've updated my fork and sync-ed, is it OK?

**Comment by gunnarmorling** [comment]
Yepp, all looking good now. Nice idea with the trie! 00:17.815. Thx for participating!

**Comment by hallvard** [comment]
Thanks for your efforts, the "competition" has created som buzz in a
positive sense!
And most of all, it has created a lot of learning opportunities!

Hallvard

**Pull request merged**